src/helpers/crypto.ts

```diff
@@ -1,20 +1,56 @@
 import { IKeyringPair } from "@polkadot/types/types"
+import { SignerResult, Signer } from "@polkadot/api/types"
 import { u8aToHex } from "@polkadot/util"
 import { getRawApi } from "../blockchain"
 import { Buffer } from "buffer"
 
 /**
- * @name getSignature
+ * @name getSignatureFromKeyring
  * @summary         Signs data using the keyring.
  * @param keyring   Account that will sign the data.
  * @param data      Data to be signed.
  * @returns         Hex value of the signed data.
  */
-export const getSignature = (keyring: IKeyringPair, data: string) => {
+export const getSignatureFromKeyring = (keyring: IKeyringPair, data: string) => {
   const finalData = new Uint8Array(Buffer.from(data))
   return u8aToHex(keyring.sign(finalData))
 }
 
+/**
+ * @name getSignatureFromExtension
+ * @summary                 Signs data using an injector extension. We recommand Polkadot extention.
+ * @param signerAddress     Account address that will sign the data.
+ * @param injectorExtension The signer method retrived from your extension: object must have a signer key.
+ * @param data              Data to be signed.
+ * @returns                 Hex value of the signed data.
+ */
+export const getSignatureFromExtension = async (
+  signerAddress: string,
+  injectorExtension: Record<string, Signer | any>,
+  data: string,
+) => {
+  // To handle Polkadot Extension
+  if (injectorExtension && "signer" in injectorExtension && injectorExtension?.signer?.signRaw) {
```

Comment:
You should be able to remove the Optional Chaining operator "?." here : `injectorExtension.signer.signRaw`

Same on the line below.